storage/src/vespa/storage/frameworkimpl/status/statuswebserver.cpp

```diff
@@ -236,37 +144,39 @@ namespace {
     };
 }
 
+
+int
+StatusWebServer::getListenPort() const
+{
+    return _httpServer ? _httpServer->getListenPort() : -1;
+}
+
 void
-StatusWebServer::handlePage(const framework::HttpUrlPath& urlpath, std::ostream& out)
+StatusWebServer::handlePage(const framework::HttpUrlPath& urlpath, vespalib::Portal::GetRequest request)
 {
     vespalib::string link(urlpath.getPath());
     if (link.size() > 0 && link[0] == '/') link = link.substr(1);
 
     size_t slashPos = link.find('/');
     if (slashPos != std::string::npos) link = link.substr(0, slashPos);
 
-    bool pageExisted = false;
     if ( ! link.empty()) {
         const framework::StatusReporter *reporter = _reporterMap.getStatusReporter(link);
         if (reporter != nullptr) {
             try {
-                pageExisted = reporter->reportHttpHeader(out, urlpath);
-                if (pageExisted) {
-                    pageExisted = reporter->reportStatus(out, urlpath);
+                std::ostringstream content;
+                auto content_type = reporter->getReportContentType(urlpath);
+                if (reporter->reportStatus(content, urlpath)) {
+                    request.respond_with_content(content_type, content.str());
+                } else {
+                    request.respond_with_error(404, "Not Found");
                 }
             } catch (std::exception &e) {
-                HttpErrorWriter writer(out, "500 Internal Server Error");
-                writer << "<pre>" << e.what() << "</pre>";
-                pageExisted = true;
-            }
-            if (pageExisted) {
-                LOG(spam, "Status finished request");
-                return;
+                request.respond_with_error(500, "Internal Server Error");
```

Comment:
Would be nice with a log warning here with the exception message to highlight any funny business going on internally

Reply:
agree